cdap-ui/app/cdap/components/Reports/ReportsServiceControl/index.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import enableDataPreparationService from 'components/DataPrep/DataPrepServiceControl/ServiceEnablerUtilities';
+import {MyArtifactApi} from 'api/artifact';
+import LoadingSVG from 'components/LoadingSVG';
+import {MyReportsApi} from 'api/reports';
+import IconSVG from 'components/IconSVG';
+
+const ReportsArtifact = 'cdap-program-report';
+
+export default class ReportsServiceControl extends Component {
+  static propTypes = {
+    onServiceStart: PropTypes.func
+  };
+
+  state = {
+    loading: false,
+    error: null,
+    artifactNotAvailable: false,
+    showEnableButton: false
+  };
+
+  componentDidMount() {
+    this.checkIfReportsIsAvailable();
+  }
+
+  checkIfReportsIsAvailable = () => {
+    // The artifact will be in the system namespace, so we are just checking the default namespace
+    // since default namespace will always be there.
+    let params = {
+      namespace: 'default',
+      scope: 'SYSTEM'
+    };
+
+    MyArtifactApi.list(params)
+      .subscribe(
+        (artifacts) => {
+          let isArtifactPresent = artifacts.find(artifact => artifact.name === ReportsArtifact);
+          if (!isArtifactPresent) {
+            this.setState({
+              artifactNotAvailable: true
+            });
+          } else {
+            this.setState({
+              showEnableButton: true
+            });
+          }
+        }
+      );
+  };
+
+  enableReports = () => {
+    this.setState({
+      loading: true
+    });
+
+    enableDataPreparationService({
```

Comment:
nit: At some point we should change the name of this function 😀 Looks super confusing when someone new comes and looks at mmds or here and wonder why dataprep service needs to be enabled.